pkg/master/pod_cache.go

```diff
@@ -192,6 +192,7 @@ func (p *PodCache) computePodStatus(pod *api.Pod) (api.PodStatus, error) {
 	} else {
 		newStatus.Info = result.Status.Info
 		newStatus.PodIP = result.Status.PodIP
+		newStatus.Conditions = result.Status.Conditions
```

Comment:
Sigh. I hate the pod cache. I don't have time to check right now, but it's worth looking about what will happen if kubelet can't be contacted.